### PR TITLE
docs(getting-started): add routing page

### PR DESCRIPTION
**Pull request by clemcode**
### 🔗 Linked issue

#6904

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR adds the Routing section to Getting Started guide:
- Pages
- Navigation
- Route parameters
- Route middleware

### CODE REVIEW

File: docs/content/1.getting-started/5.routing.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Routing
+
+One of Nuxt core feature is the file-system router. Every Vue file created inside the pages/ directory creates a corresponding URL (or route) that displays the content of the file.
+
+## Pages
+
+Nuxt routing is based on Vue-router. Under the hood, Nuxt generates the vue-router configuration file with every components created in the pages directory, based on their file name.
```
Comment (pi0) [inline docs/content/1.getting-started/5.routing.md:7]: Would be nice to have more external links for readers.
```suggestion
Nuxt routing is based on [vue-router](https://router.vuejs.org/). Under the hood, Nuxt generates the vue-router configuration file with every components created in the pages directory, based on their file name.
```

File: docs/content/1.getting-started/5.routing.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  ]
+}
+```
+
+::
+
+## Navigation
+
+The `<NuxtLink>` component links pages between them. It renders a `<a>` tag with the `href` attribute set to the route of the page. Once the application is hydrated, pages transitions are performed in JavaScript by updating the browser URL. This prevents full-page refreshes and allow for animated transitions.
```
Comment (pi0) [inline docs/content/1.getting-started/5.routing.md:39]: We _could_ introduce prefetching feature too! It just landed for RC.10

File: docs/content/1.getting-started/5.routing.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+::alert{type=info}
+Route middleware run within the Vue part of your Nuxt app. Despite the similar name, they are completely different from server middleware, which are run in the Nitro server part of your app.
+::
+
+There are three kinds of route middleware:
+
+1. Anonymous (or inline) route middleware, which are defined directly in the pages where they are used.
+2. Named route middleware, which are placed in the `middleware/` directory and will be automatically loaded via asynchronous import when used on a page. (**Note**: The route middleware name is normalized to kebab-case, so `someMiddleware` becomes `some-middleware`.)
+3. Global route middleware, which are placed in the `middleware/` directory (with a `.global` suffix) and will be automatically run on every route change.
```
Comment (pi0) [inline docs/content/1.getting-started/5.routing.md:82]: Maybe add an example for middleware?

File: docs/content/1.getting-started/5.routing.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
+# Routing
+
+One of Nuxt core feature is the file-system router. Every Vue file created inside the pages/ directory creates a corresponding URL (or route) that displays the content of the file.
```
Comment (Atinux) [inline docs/content/1.getting-started/5.routing.md:3]:
```suggestion
One of Nuxt core feature is the file-system router. Every Vue file created inside the pages/ directory creates a corresponding URL (or route) that displays the content of the file. It also code-split every page to a separate chunk by leveraging dynamic imports to ship the minimum of JavaScript for the requested route.
```

File: docs/content/1.getting-started/5.routing.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+# Routing
+
+One of Nuxt core feature is the file-system router. Every Vue file created inside the pages/ directory creates a corresponding URL (or route) that displays the content of the file. Nuxt leverages code-splitting on each page by using dynamic imports to ship the minimum of JavaScript for the requested route.
+
+## Pages
+
+Nuxt routing is based on [vue-router](https://router.vuejs.org/). Under the hood, Nuxt generates the vue-router configuration file with every components created in the pages directory, based on their file name.
```
Comment (Atinux) [inline docs/content/1.getting-started/5.routing.md:7]:
```suggestion
Nuxt routing is based on [vue-router](https://router.vuejs.org/) and generates the routes configuration with every components created in the [`pages/`](/guide/directory-structure/pages) directory, based on their file name.
```

File: docs/content/1.getting-started/5.routing.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+```vue [pages/post/[id].vue]
+<script setup>
+const route = useRoute()
+
+// When accessing /posts/1, route.params.id will be 1
+console.log(route.params.id)
+</script>
+```
+
```
Comment (Atinux) [inline docs/content/1.getting-started/5.routing.md:69]:
```suggestion
:ReadMore{link="/api/composables/use-route"}

```

File: docs/content/1.getting-started/5.routing.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+::alert{type=info}
+Route middleware run within the Vue part of your Nuxt app. Despite the similar name, they are completely different from server middleware, which are run in the Nitro server part of your app.
+::
+
+There are three kinds of route middleware:
+
+1. Anonymous (or inline) route middleware, which are defined directly in the pages where they are used.
+2. Named route middleware, which are placed in the `middleware/` directory and will be automatically loaded via asynchronous import when used on a page. (**Note**: The route middleware name is normalized to kebab-case, so `someMiddleware` becomes `some-middleware`.)
+3. Global route middleware, which are placed in the `middleware/` directory (with a `.global` suffix) and will be automatically run on every route change.
```
Comment (Atinux) [inline docs/content/1.getting-started/5.routing.md:83]:
```suggestion
3. Global route middleware, which are placed in the `middleware/` directory (with a `.global` suffix) and will be automatically run on every route change.

Example of an `auth` middleware protecting the `/dashboard` page:

::code-group

```ts [middleware/auth.ts]
export default defineNuxtRouteMiddleware((to, from) => {
  if (isAuthenticated() === false) {
    return navigateTo('/login')
  }
})
```

```html [pages/dashboard.vue]
<script setup>
definePageMeta({
  middleware: 'auth'
})
</script>

<template>
  <h1>Welcome to your dashboard</h1>
</template>
```

::

:ReadMore{link="/guide/directory-structure/middleware"}
```

File: docs/content/1.getting-started/5.routing.md (hunk 2 above)
Comment (Atinux) [reply docs/content/1.getting-started/5.routing.md:39]: Happy to make another PR once RC10 is landed?

**Review by Atinux — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**